packages/@uppy/core/src/index.js

```diff
@@ -1564,8 +1572,24 @@ class Uppy {
         return
       }
 
-      const files = currentUpload.fileIDs
-        .map((fileID) => this.getFile(fileID))
+      // Mark postprocessing step as complete if necessary; this addresses a case where we might get
+      // stuck in the postprocessing UI while the upload is fully complete.
+      // If the postprocessing steps do not do any work, they may not emit postprocessing events at
+      // all, and never mark the postprocessing as complete. This is fine on its own but we
+      // introduced code in the @uppy/core upload-success handler to prepare postprocessing progress
+      // state if any postprocessors are registered. That is to avoid a "flash of completed state"
+      // before the postprocessing plugins can emit events.
+      //
+      // So, just in case an upload with postprocessing plugins *has* completed *without* emitting
+      // postprocessing completion, we do it instead.
```

Comment:
To re-iterate, just to make sure I understand and the comment here is sufficient.
This re-emits `postprocess-complete` after `lastStep` of the upload/process chain, in case the event wasn’t emitted by processing plugins? 

I’m missing how this is related to websockets and polling fallback. When falling back to polling we missed `postprocess-complete`?

Reply:
Right. If the upload has completed, but the postprocessing step was not marked complete, we can assume that it actually is--otherwise the entire upload would not have completed yet.

I think this may just be a UI bug that happened to show up at the same time rather than a polling bug, but I'm not really sure 😅